pallets/stellar-relay/src/lib.rs

```diff
@@ -565,27 +584,60 @@ pub mod pallet {
 
 				ensure!(node_id_found, Error::<T>::EnvelopeSignedByUnknownValidator);
 
+				// Check if all envelopes are using the same slot index
+				ensure!(
+					slot_index == envelope.statement.slot_index,
+					Error::<T>::EnvelopeSlotIndexMismatch
+				);
+
 				let signature_valid = verify_signature(envelope, &node_id, network);
 				ensure!(signature_valid, Error::<T>::InvalidEnvelopeSignature);
 			}
 
 			// Check if transaction set matches tx_set_hash included in the ScpEnvelopes
 			let expected_tx_set_hash = compute_non_generic_tx_set_content_hash(transaction_set)
-				.ok_or(Error::<T>::FailedToComputenonGenericTxSetContentHash)?;
+				.ok_or(Error::<T>::FailedToComputeNonGenericTxSetContentHash)?;
+
+			// We store the externalized value in a variable so that we can check if it's the same
+			// for all envelopes. We don't distinguish between externalized and confirmed values as
+			// it should be the same value regardless.
+			let mut externalized_value: Option<Value> = None;
```

Comment:
We can also initialize these two values from `externalized_envelope` so that they don't need to be optional – simplifies the equality check logic below a little.

Reply:
Right, I'll change that

Reply:
I simplified that logic in [this](https://github.com/pendulum-chain/spacewalk/pull/362/commits/2ac84397528cc74eeac4c3f8c7dc0cbe173564a8) commit